packages/app-project/src/screens/SubjectTalkPage/components/SubjectTalkData/components/Tags/Tags.jsx

```diff
@@ -0,0 +1,204 @@
+  &::after {
+    content: "";
+    flex: 1 1 auto;
+    height: 1px;
+    background: ${props => props.theme.dark ? props.theme.global.colors['accent-1'] : props.theme.global.colors['neutral-1']};
+  }
+`
+
+const StyledAddTagButton = styled(Button)`
+  &:focus,
+  &:hover {
+    text-decoration: underline;
+  }
+
+  &:disabled {
+    cursor: not-allowed;
+    text-decoration: none;
+  }
+`
+
+const DEFAULT_HANDLER = () => true
+
+function Tags({
+  error = undefined,
+  loading = false,
+  onTagClick = DEFAULT_HANDLER,
+  tags = undefined,
+  userId = undefined
+}) {
+  const { t } = useTranslation('screens')
+
+  const tagHeading = tags?.length > 0 ? t('Talk.tags') : t('Talk.noTags')
+
+  return (
+    <Box
+      gap='small'
+      height={{ min: 'auto' }}
+      pad='small'
+    >
+      <Box
+        direction='row'
+        justify='between'
+      >
+        <Box
+          align='center'
+          direction='row'
+          gap='xsmall'
+        >
+          <TagIcon />
+          <StyledHeading
+            color={{ dark: 'light-1', light: 'dark-4' }}
+            level={3}
+            size='1rem'
+          >
+            {tagHeading}
+          </StyledHeading>
+        </Box>
+        <Box
+          align='center'
+          direction='row'
+          gap='xsmall'
+        >
+          {(tags?.length > 0 && userId) ? (
+            <StyledAddTagButton
+              label={(
+                <Box
+                  align='center'
+                  direction='row'
+                  gap='xsmall'
+                  justify='center'
+                >
+                  <TagIcon
+                    color={{ dark: 'accent-1', light: 'neutral-1' }}
+                    size='16px'
+                  />
+                  <SpacedText
+                    color={{ dark: 'accent-1', light: 'neutral-1' }}
+                    size='16px'
+                  >
+                    {t('Talk.addATag')}
+                  </SpacedText>
+                </Box>
+              )}
+              margin={{ horizontal: 'xsmall' }}
+              onClick={() => window.alert('coming soon!')}
+              plain
+            />
+          ) : tags?.length > 0 ? (
+            <SpacedText>
+              {t('Talk.logInToTag')}
+            </SpacedText>
+          ) : null}
+        </Box>
+      </Box>
+      {error ? (
+        <Box align='center' justify='center' fill pad='medium'>
+          <SpacedText uppercase={false}>
+            {error?.message}
+          </SpacedText>
+        </Box>
+      ) : loading ? (
+        <Box align='center' justify='center' fill pad='medium'>
+          <Loader />
+        </Box>
+      ) : tags?.length > 0 ? (
+        <StyledOrderedList
+          forwardedAs='ol'
+          direction='row'
+          margin='none'
+          pad='none'
+          wrap
+        >
+          {tags?.map(tag => (
+            <li
+              key={`${tag.id}-${tag.name}`}
+              style={{ margin: '0 10px 10px 0' }}
```

Comment:
```suggestion
```

See related note in this file about row-gap and column-gap. Please don't use `style`.